AlfiyaSiddique/TastyTrails
# Added Feedback Management Section to the Profile Page

### Pull Request: Add Feedback Section to Profile Page

**Description:**
This PR introduces a new feature to the user profile page that allows users to view and delete their feedback. Users can now manage their feedback directly from their profile, enhancing user experience and engagement.

---

**Changes Made:**

- **Frontend:**
  - Updated the profile page layout to include a dedicated feedback section displaying all feedback entries associated with the user.
  - Implemented functionality to fetch user feedbacks using the new API.
  - Added delete buttons for each feedback entry, allowing users to remove their feedback easily.

- **Backend:**
  - Added new API endpoints:
    - `GET /api/feedback/:id`: Retrieves all feedback entries written by the user identified by the given `id`.
    - `POST /api/feedback/delete`: Deletes a specific feedback entry based on the provided feedback ID.
  - Implemented the necessary controllers to handle the logic for retrieving and deleting feedback based on user ID and feedback ID.

---

**How to Test:**
1. Navigate to the profile page of a user.
2. Verify that the feedback section displays all feedback entries associated with the user.
3. Test the delete functionality by clicking the delete button on a feedback entry. Confirm that the feedback is removed from the list.

### Video:

https://github.com/user-attachments/assets/b0b7b271-84ff-4530-ab11-cf17ec5d79aa

## Checklist
